tests/s390x_tests/TOOL_s390_vmcp/consoletest_vmcp.pm

```diff
@@ -0,0 +1,104 @@
+#
+# Copyright © 2009-2013 Bernhard M. Wiedemann
+# Copyright © 2012-2016 SUSE LLC                                                                                    
+# Copyright (C) 2018 IBM Corp.
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+#
+# Summary:  Based on consoltest_setup.pm (console test pre setup, stopping and disabling packagekit, install curl and tar to get logs and so on)
+# modified for running the testcase TOOL_s390_vmcp on s390x.
+
+use base "consoletest";
+use testapi;
+use utils;
+use strict;
+
+sub run {
+    my $self = shift;
+    # let's see how it looks at the beginning
+    save_screenshot;
+
+    # Special keys like Ctrl-Alt-Fx does not work on Hyper-V atm. Alt-Fx however do.
+    my $tty1_key = 'ctrl-alt-f1';
+    if (check_var('VIRSH_VMM_FAMILY', 'hyperv')) {
+        $tty1_key = 'alt-f1';
+    }
+
+    if (!check_var('ARCH', 's390x')) {
+        # verify there is a text console on tty1
+        for (1 .. 6) {
+            send_key $tty1_key;
+            if (check_screen("tty1-selected", 5)) {
+                last;
+            }
+        }
+        if (!check_screen "tty1-selected", 5) {    #workaround for bsc#977007
+            record_soft_failure "unable to switch to the text mode";
+            send_key 'ctrl-alt-backspace';         #kill X and log in again
+            send_key 'ctrl-alt-backspace';
+            assert_screen 'displaymanager', 200;    #copy from installation/first_boot.pm
+            mouse_hide();
+            if (get_var('DM_NEEDS_USERNAME')) {
+                type_string $username;
+            }
+            if (match_has_tag("sddm")) {
+                # make sure choose plasma5 session
+                assert_and_click "sddm-sessions-list";
+                assert_and_click "sddm-sessions-plasma5";
+                assert_and_click "sddm-password-input";
+            }
+            else {
+                wait_screen_change { send_key 'ret' };
+            }
+            type_string "$password";
+            send_key "ret";
+            send_key_until_needlematch "tty1-selected", $tty1_key, 6, 5;
+        }
+    }
+
+    # init
+    check_console_font;
+    type_string "chown $username /dev/$serialdev\n";
+    script_run 'echo "set -o pipefail" >> /etc/bash.bashrc.local';
+    script_run '. /etc/bash.bashrc.local';
+    # Export the existing status of running tasks and system load for future reference (fail would export it again)
+    script_run "ps axf > /tmp/psaxf.log";
+    script_run "cat /proc/loadavg > /tmp/loadavg_consoletest_setup.txt";
+
+    # Just after the setup: let's see the network configuration
+    save_screenshot;
+
+    $self->clear_and_verify_console;
+
+
+# start of modification for s390 test TOOL_s390_vmcp    
+    my $TC_PATH = get_var('TC_PATH') . "/TOOL_s390_vmcp/";
```

Comment:
to be honest, I don't quite get what to set this variable to.
Was it some webserver you provided internally where you fetched the scripts from?
it might be worth to take a look on https://github.com/os-autoinst/os-autoinst/blob/master/testapi.pm#L1779

Reply:
We don’t know other means of transferring the test scripts to the target z/VM guest. Could you please suggest us the correct way of putting them? Is there preinstalled web server on openQA host which we can use? Or can we use “git fetch"?

Reply:
@elifaln if you put the data in the data directory of the tests, you can download them later via data_url($filename) call, like they're doing it in the [console/openvswitch_ssl](https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/tests/console/openvswitch_ssl.pm#L41) test

Reply:
About `data_url`. We've tried the function and it provides some sophisticated (seems as one-time) URL, but we cannot figure out where actually to place our scripts into so they will be available via the provided URL? 

Could you please help us to figure out, which folder on an openQA server we must put our scripts into?

Reply:
Hi @timbgr 

so data_url() basically opens a connection on the worker from where the SUT (System under Test) can receive files from. the url contains a unique string bound to the job afaik.
The connection always points to the `data/` subdirectory of `os-autoinst-distri-opensuse`, so for example:

data_url(''): http://$worker:20993/C5mJaCtD3xFwm6YU/data/ points to https://github.com/os-autoinst/os-autoinst-distri-opensuse/tree/master/data
data_url('autoyast'): http://$worker:20993/C5mJaCtD3xFwm6YU/data/autoyast points to https://github.com/os-autoinst/os-autoinst-distri-opensuse/tree/master/data/autoyast


from here, the system under test could receive files e.g. via `curl` or `wget`.

Hope that answers your questions

Reply:
I modified consoletest_vmcp.pm so now it's using data_url() function to fetch the shell scripts to be run.
`   script_run "mkdir TOOL_s390_vmcp";`
`    script_run "cd TOOL_s390_vmcp";`
`    assert_script_run "wget " . data_url('s390-tests/vmcp_main.sh');`
`    assert_script_run "wget " . data_url('s390-tests/common.sh');`
It's working fine :+1: